### ISSUE TITLE
Agentic GRPO with vanilla rollout never learns by default: every sampler call uses PRNGKey(0), so all generations in a group are identical

### ISSUE THREAD

**Issue by howl-anderson** [google/tunix#2398]
## Summary

With the vanilla rollout engine, agentic GRPO does not learn with the default `RolloutConfig`: all `num_generations` samples of a prompt come out token-identical. Every group then has zero reward variance, every advantage is 0, and the policy never changes. Nothing errors or warns, and the reward curve still moves (it just tracks which prompts land in which step), so a run can look like training when it isn't.

## Cause

- `RolloutConfig.seed` defaults to `None` (`tunix/common/configs.py`).
- `VanillaRollout.generate` passes it straight through: `seed=rollout_config.seed` (`tunix/rl/rollout/vanilla_rollout.py`).
- `Sampler.__call__` turns `None` into `jax.random.PRNGKey(0)` on every call (`tunix/generate/sampler.py`, `if seed is None: seed = jax.random.PRNGKey(0)`), and per-step keys are `fold_in(seed, decoding_step)`.
- The agentic pipeline samples each episode in its own call, so the same prompt with the same key produces the same tokens, regardless of `temperature` / `top_p`.

Checked against `main` (20c2baa): the code paths above are unchanged.

## Evidence

Qwen3-0.6B, `agentic_grpo_learner.GRPOLearner`, `ToolAgent` + `ToolEnvironment`, `rollout_engine="vanilla"`, `num_generations=4`, `temperature=0.7`, `top_p` default, `seed` default. TPU v5e-8 on Kaggle. We logged every episode's final action and grouped by prompt:

| run | groups | groups with 4 identical actions | groups with any reward variance |
|---|---|---|---|
| default seed, lr 1e-6, 50 steps | 400 | 400 | 0 |
| default seed, lr 4e-6, 100 steps | 800 | 800 | 0 |
| default seed, lr 2e-6, beta 0.1, 100 steps | 800 | 800 | 0 |
| fresh seed per call (below), lr 1e-6, 100 steps | 800 | 440 | 52 |

With the default seed, per-prompt accuracy was strictly 0 or 1 (618 prompts always right, 182 always wrong), and a run with the reward accidentally pinned to 0 (so zero gradient by construction) produced the same curve as a real run. After the fix, the policy trained: mean reward went from 0.77 (untrained) to 0.996 over steps 25–100, and on the 182 prompts the untrained model always failed, accuracy went from 0.73 early in training to 0.997 late.

## Workaround

```python
from tunix.generate import sampler as sampler_lib

_orig_call = sampler_lib.Sampler.__call__
_calls = [0]

def _fresh_seed_call(self, *args, **kwargs):
    if kwargs.get("seed") is None:
        _calls[0] += 1
        kwargs["seed"] = _calls[0]
    return _orig_call(self, *args, **kwargs)

sampler_lib.Sampler.__call__ = _fresh_seed_call
```

## Possible fixes

- Have `VanillaRollout` (or `Sampler` when `seed is None`) derive a fresh key per call, e.g. from a counter or by splitting a stored key.
- Or have the agentic learner pass a distinct seed per episode.
- At least warn when `num_generations > 1` and every sample in a group is identical, since that silently zeroes the GRPO signal.

Happy to send a PR for whichever direction you prefer.
